docs/porting/psa/TLS-on-PSA.md

```diff
@@ -0,0 +1,114 @@
+# Using PSA-enabled Mbed TLS
+
+### Building PSA-enabled Mbed TLS from Git
+
+The version of Mbed TLS shipped in Mbed OS builds with PSA enabled by default. However, you can build a PSA-enabled version of Mbed TLS from [our Git repository](https://github.com/ARMmbed/mbedtls):
+
+1. Clone our repository, switch to the `development-psa` branch and initialize the submodule: `git checkout development-psa && git submodule update --init`.
+1. Enable the `MBEDTLS_USE_PSA_CRYPTO` build-time configuration option in `include/mbedtls/config.h`, either by editing the file manually, or using the config script: `scripts/config.pl set MBEDTLS_USE_PSA_CRYPTO`.
+1. Build normally, for example (with GNU Make) make all test or (with CMake) `mkdir build && cd build && cmake .. && make all test`.
+
+### Switching from pre-PSA Mbed TLS to PSA-enabled Mbed TLS
+
+You can use applications built with pre-PSA Mbed TLS with PSA-enabled Mbed TLS by recompiling them against a PSA-enabled version of Mbed TLS. (Note that relinking is not enough, though.) In other words, PSA-enabled Mbed TLS is currently fully API-compatible with non-PSA-enabled Mbed TLS, but not ABI compatible. All existing (pre-PSA) APIs are preserved and continue to exist in parallel with new PSA-based APIs, to ease the transition for users at this stage.
+
+_Please note that this is likely to change at a later date though, as existing interfaces are likely to be deprecated in favor of the new, PSA-based APIs - if, when and how it happens is outside the scope of this document._
+
+### Switching from PSA-enabled Mbed TLS in Mbed OS 5.11 to PSA-enabled Mbed TLS in Mbed OS 5.12
```

Comment:
I don't want this to be a document just about going from 5.11 to 5.12. That narrows it to being relevant to only one quarter. Could we instead position it as how to enable PSA in Mbed TLS, without mentioning version numbers?